core/matrix/dense.cpp

```diff
@@ -238,6 +249,40 @@ void Dense<ValueType>::add_scaled_impl(const LinOp* alpha, const LinOp* b)
 }
 
 
+template <typename ValueType>
+void Dense<ValueType>::add_scale(const LinOp* const alpha, const LinOp* const a,
+                                 const LinOp* const beta)
+{
+    GKO_ASSERT_EQUAL_ROWS(alpha, dim<2>(1, 1));
+    GKO_ASSERT_EQUAL_ROWS(beta, dim<2>(1, 1));
+    if (alpha->get_size()[1] != 1) {
+        // different alpha/beta for each column
+        GKO_ASSERT_EQUAL_COLS(this, alpha);
+        GKO_ASSERT_EQUAL_COLS(this, beta);
+    }
+    GKO_ASSERT_EQUAL_DIMENSIONS(this, a);
+    auto exec = this->get_executor();
+    auto alphac = make_temporary_clone(exec, alpha);
+    auto betac = make_temporary_clone(exec, beta);
+    auto ac = make_temporary_clone(exec, a);
+    if (dynamic_cast<const ConvertibleTo<Dense<>>*>(alpha) &&
+        is_complex<ValueType>()) {
+        exec->run(dense::make_add_scale(
+            make_temporary_conversion<remove_complex<ValueType>>(alphac.get())
```

Comment:
Why do we need to have `alpha` and `beta` of non-complex type ?

Reply:
I guess we don't need that functionality. But I thought I'll keep it similar to the `add_scaled` function in `Dense`.